src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -59,7 +60,8 @@ private async Task CleanOutputDirectory(CancellationToken cancellationToken)
             foreach (var subDir in Directory.EnumerateDirectories(config.OutputPath))
                 Directory.Delete(subDir, true);
             await lockManagementService.BackupLockFileAsync(config.OutputPath, cancellationToken);
-            foreach (var subFile in Directory.EnumerateFiles(config.OutputPath))
+            foreach (var subFile in Directory.EnumerateFiles(config.OutputPath)
```

Comment:
on kiota update --clean-output, i think the lock file will get moved from the folder here so the update wouldn't occur.

Reply:
just tested this scenario and it seems to be working fine with that PR.
when updating:
1. the CLI finds all kiota lock files, loads settings from them, and triggers multiple generations.
2. during each generation, the clean output is applied.
3. during each generation, evaluation of whether kiota should generate or skip always returns true on clean output.

https://github.com/microsoft/kiota/blob/0dea355895ea1643095386f2e569b8155717a6f1/src/kiota/Handlers/KiotaUpdateCommandHandler.cs#L50

https://github.com/microsoft/kiota/blob/0dea355895ea1643095386f2e569b8155717a6f1/src/Kiota.Builder/KiotaBuilder.cs#L53

https://github.com/microsoft/kiota/blob/0dea355895ea1643095386f2e569b8155717a6f1/src/Kiota.Builder/KiotaBuilder.cs#L136